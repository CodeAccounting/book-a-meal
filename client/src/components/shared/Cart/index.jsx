import React, { Fragment, Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import LinkBtn from '../Link';
import { orderItemPropTypes } from '../../../helpers/proptypes';
import checkShopOpen from '../../../helpers/checkShopOpen';
import calculateCashEarnedFromOrder from '../../../helpers/calculateCashEarnedFromOrder';

/**
 * @exports
 * @extends Component
 * @class Cart
 * @returns {JSX} Cart
 */
class Cart extends Component {
  static propTypes = {
    order: PropTypes.arrayOf(orderItemPropTypes).isRequired,
    handleQuantityInputChange: PropTypes.func.isRequired,
    removeOrderItem: PropTypes.func.isRequired
  }

  state = {
    show: false
  }

  /**
   * @memberof Cart
   * @returns {JSX} Cart Component
   */
  toggleCart = () => {
    this.setState(prevState => ({
      show: !prevState.show
    }));
  }

  /**
   * @memberof Cart
   * @returns {JSX} Checkout Btn
   */
  renderCheckoutBtn = () => (
    <Link to="/order-confirmation" href="/order-confirmation">
      <button className="btn btn-pri btn-block checkout-btn">Checkout</button>
    </Link>);

  /**
   * @memberof Cart
   * @returns {JSX} Order Item
   */
  renderOrderSummary = () => {
    const {
      order, handleQuantityInputChange, removeOrderItem
    } = this.props;

    return (
      <div className="order-summary">
        {order.map(item => (
          <div key={item.id}>
            <p>
              <input type="number" className="order-input" value={item.quantity} min="1" onChange={e => handleQuantityInputChange(e, item.id)} />
            </p>
            <p>{item.title}</p>
            <p>{item.price}</p>
            <LinkBtn className="remove-order" clickHandler={() => removeOrderItem(item.id)}>&times;</LinkBtn>
          </div>
        ))}
        <div className="order-amount">
          <div>
            <p>Total:</p>
            <h2>&#8358;{calculateCashEarnedFromOrder(order)}</h2>
          </div>
        </div>
      </div>
    );
  }

  /**
   * @memberof Cart
   * @returns {JSX} Cart Toggle
   */
  renderCartToggle = () => {
    const { order } = this.props;
    const isShopOpen = checkShopOpen();

    return (
      <div className="side-cart-toggle d-none-xl">
        <span aria-hidden="true" id="cart-toggler" onClick={this.toggleCart}>
          <div className="text-center">
            <img src="/img/cart.png" alt="cart" /> &nbsp;
            <sup>{order.length}</sup>
          </div>
          <p className="text-center">Total: &#8358;{calculateCashEarnedFromOrder(order)}</p>
        </span>
        {isShopOpen &&
          <span className="cart-btn">
            {this.renderCheckoutBtn()}
          </span>}
      </div>
    );
  }

  /**
   * @memberof Cart
   * @returns {JSX} Cart Component
   */
  render() {
    const { order } = this.props;
    const isShopOpen = checkShopOpen();

    return (
      <Fragment>
        {this.renderCartToggle()}
        <div className={`side-cart ${this.state.show && 'show'}`} id="side-cart">
          <div className="cart">
            <div className="cart-header">
              <h2>Your Cart</h2>
              <div className="close d-none-xl">
                <LinkBtn aria-hidden="true" id="side-cart-close-icon" clickHandler={this.toggleCart}>&times;</LinkBtn>
              </div>
            </div>
            <hr />
            {order.length === 0 &&
              <div className="empty-cart">
                <p>Your Cart is Empty</p>
              </div>}
            {order.length !== 0 && this.renderOrderSummary()}
            <br />
            {isShopOpen && this.renderCheckoutBtn()}
          </div>
        </div>
      </Fragment>
    );
  }
}

export default Cart;
