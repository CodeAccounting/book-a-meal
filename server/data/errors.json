{
  "401": "Unauthorized",
  "403": "Forbidden",
  "404": "Not Found",
  "422": "Validation Error",
  "500": "Internal Server Error",
  "empty": "Empty PUT Requests Are Not Allowed"
}