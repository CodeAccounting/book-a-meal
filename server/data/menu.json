[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "date": "2018-05-06",
     "meals": [
        "81211c24-51c0-46ec-b1e0-18db55880958",
        "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
        "baa0412a-d167-4d2b-b1d8-404cb8f02631",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "date": "2018-04-06",
     "meals": [
        "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
        "3ec802c6-0a32-4d29-b27b-42e0f4b532dd"
     ],
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "date": "2018-05-07",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "date": "2018-04-06",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614",
        "46ced7aa-eed5-4462-b2c0-153f31589bdd"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
     "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
     "date": "2018-04-27",
     "meals": [
        "91b6e41c-0972-4ac5-86da-4ac1f5226e83",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3",
        "8a65538d-f862-420e-bcdc-80743df06578"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  }
]