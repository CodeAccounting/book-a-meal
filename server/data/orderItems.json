[
  {
     "id": 1,
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 2,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
     "id": 2,
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 1,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  }, 
  {
    "id": 3,
    "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
    "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
    "quantity": 2,
    "createdAt": "2018-04-06T14:34:90.000",
    "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
    "id": 4,
    "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  },
  {
    "id": 5,
    "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  }
]