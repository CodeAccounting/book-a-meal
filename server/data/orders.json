[
  {
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "ce228787-f939-40a0-bfd3-6607ca8d2e53",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e544248c-145c-4145-b165-239658857637",
     "created": "2018-05-07T14:34:90.000",
     "updated": "2018-05-07T14:34:90.000",
     "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "e5508b87-7975-493d-a900-3d47a69dad03",
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000",
     "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  },
  {
     "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000",
     "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "deliveryAddress": "4, Church Street, Yaba",
     "deliveryPhoneNo": "+2348134567890",
     "quantity": 2
  }
]