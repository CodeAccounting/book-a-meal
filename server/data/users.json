[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
    "username": null,
    "businessName": "FoodCircle",
    "email": "food@circle.com",
    "password": "foodcircle",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": "+2348134567890",
    "businessAddress": "4, Church Street, Yaba",
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "caterer"
  },
  {
    "userId": "61bb8f8d-3b59-4294-acbc-16623818c391",
    "username": "Iveren",
    "businessName": null,
    "email": "iveren@shaguy.com",
    "password": "iverenshaguy",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": null,
    "businessAddress": null,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "user"
  },
  {
     "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
     "username": "Olisa",
     "businessName": null,
     "email": "olisa@emodi.com",
     "password": "olisaemodi",
     "passwordHash": "23456789dtfgyuhio567890",
     "businessPhone": null,
     "businessAddress": null,
     "createdAt": "2018-05-06T14:34:90.000",
     "updatedAt": "2018-05-06T14:34:90.000",
     "role": "user"
  },
  {
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "username": "Emiola",
     "businessName": null,
     "email": "emiola@olasanmi.com",
     "password": "emiolaolasanmi",
     "passwordHash": "23456789dtfgyuhio567890",
     "businessPhone": null,
     "businessAddress": null,
     "createdAt": "2018-05-06T14:34:90.000",
     "updatedAt": "2018-05-06T14:34:90.000",
     "role": "user"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
    "username": null,
    "businessName": "BellyFill",
    "email": "belly@fill.com",
    "password": "bellyfil",
    "passwordHash": "23456789dtfgyuhio567890",
    "businessPhone": "+2348134567890",
    "businessAddress": "4, Church Street, Yaba",
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000",
    "role": "caterer"
  }
]