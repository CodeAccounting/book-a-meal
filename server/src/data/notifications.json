[
  {
    "notifId": "3dc9cc9b-179b-4f0d-ac60-7ca6f4659d79",
    "userId": "61bb8f8d-3b59-4294-acbc-16623818c391",
    "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
    "message": "Jollof Spaghetti, Plantain and Turkey was just added to the menu",
    "isRead": true,
    "created": "2018-05-06T14:34:90.000",
    "updated": "2018-05-06T14:34:90.000"
 },
  {
    "notifId": "87a311b7-898b-4a6b-81c2-619417c96432",
    "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
    "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
    "message": "Jollof Spaghetti, Plantain and Turkey was just added to the menu",
    "isRead": true,
    "created": "2018-05-06T14:34:90.000",
    "updated": "2018-05-06T14:34:90.000"
 },
  {
    "notifId": "9015c150-6079-4be6-a240-272a4de6bd17",
    "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
    "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
    "message": "Jollof Spaghetti, Plantain and Turkey was just added to the menu",
    "isRead": false,
    "created": "2018-05-06T14:34:90.000",
    "updated": "2018-05-06T14:34:90.000"
 },
  {
    "notifId": "38eb0942-38b7-4a16-91e9-8b1662175d75",
    "userId": "e42c7ce0-958e-4368-adcb-e5d97dbbe5b7",
    "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
    "message": "Jollof Spaghetti, Plantain and Turkey was just added to the menu",
    "isRead": false,
    "created": "2018-05-06T14:34:90.000",
    "updated": "2018-05-06T14:34:90.000"
 },
  {
    "notifId": "d886ce22-9b41-4cf6-9dd6-7bdd5476d6c0",
     "userId": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
     "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
     "message": "Jollof Spaghetti, Plantain and Turkey was just added to the menu",
     "isRead": false,
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  }
]