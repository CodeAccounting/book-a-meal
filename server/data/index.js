import users from './users.json';
import meals from './meals.json';
import menu from './menu.json';
import orders from './orders.json';
import errors from './errors.json';

export default {
  users, meals, menu, orders, errors
};
