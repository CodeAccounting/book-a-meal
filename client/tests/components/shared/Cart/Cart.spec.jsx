import React from 'react';
import moment from 'moment';
import Cart from '../../../../src/components/shared/Cart';
import { order } from '../../../setup/data';

const { now } = Date;
const currentDay = moment().format('YYYY-MM-DD');

describe('Cart', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => new Date(currentDay).getTime() + (60 * 60 * 13 * 1000));
  });

  afterAll(() => {
    Date.now = now;
  });

  it('renders correctly', () => {
    const shallowWrapper = shallow(<Cart
      order={order}
      handleQuantityInputChange={jest.fn()}
      removeOrderItem={jest.fn()}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders message when there is no order', () => {
    const shallowWrapper = shallow(<Cart
      order={[]}
      handleQuantityInputChange={jest.fn()}
      removeOrderItem={jest.fn()}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('div.empty-cart>p').text()).toEqual('Your Cart is Empty');
  });

  it('toggles cart', () => {
    const wrapper = mount(<Cart
      order={order}
      handleQuantityInputChange={jest.fn()}
      removeOrderItem={jest.fn()}
    />, rrcMock.get());

    wrapper.find('#cart-toggler').simulate('click');

    expect(wrapper.state().show).toBeTruthy();
  });

  it('shows checkout button when shop is open', () => {
    const wrapper = mount(<Cart
      order={order}
      handleQuantityInputChange={jest.fn()}
      removeOrderItem={jest.fn()}
    />, rrcMock.get());

    expect(wrapper.find('.checkout-btn').length).toEqual(2); // 2 buttons for differet screen sizes
  });

  it('hides checkout button when shop is closed', () => {
    const dateNowSpy = jest.spyOn(Date, 'now').mockImplementation(() => new Date(currentDay).getTime() + (60 * 60 * 18 * 1000));
    const wrapper = mount(<Cart
      order={order}
      handleQuantityInputChange={jest.fn()}
      removeOrderItem={jest.fn()}
    />, rrcMock.get());

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.checkout-btn').length).toEqual(0);

    dateNowSpy.mockReset();
    dateNowSpy.mockRestore();
  });
});
