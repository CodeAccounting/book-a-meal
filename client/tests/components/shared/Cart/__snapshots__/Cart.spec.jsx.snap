// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart hides checkout button when shop is closed 1`] = `
<Cart
  handleQuantityInputChange={[MockFunction]}
  order={
    Array [
      Object {
        "id": "81211c24-51c0-46ec-b1e0-18db55880958",
        "price": "1500.00",
        "quantity": 1,
        "title": "Jollof Rice, Beef and Plantain",
      },
    ]
  }
  removeOrderItem={[MockFunction]}
>
  <div
    className="side-cart-toggle d-none-xl"
  >
    <span
      aria-hidden="true"
      id="cart-toggler"
      onClick={[Function]}
    >
      <div
        className="text-center"
      >
        <img
          alt="cart"
          src="/img/cart.png"
        />
          
        <sup>
          1
        </sup>
      </div>
      <p
        className="text-center"
      >
        Total: ₦
        1500
      </p>
    </span>
  </div>
  <div
    className="side-cart null"
    id="side-cart"
  >
    <div
      className="cart"
    >
      <div
        className="cart-header"
      >
        <h2>
          Your Cart
        </h2>
        <CloseIcon
          btnID="side-cart-close-icon"
          clickHandler={[Function]}
          divClass="d-none-xl"
        >
          <div
            className="close d-none-xl"
          >
            <Link
              aria-hidden="true"
              className={null}
              clickHandler={[Function]}
              id="side-cart-close-icon"
            >
              <button
                className="link-btn null"
                id="side-cart-close-icon"
                onClick={[Function]}
              >
                ×
              </button>
            </Link>
          </div>
        </CloseIcon>
      </div>
      <hr />
      <div
        className="order-summary"
      >
        <div
          key="81211c24-51c0-46ec-b1e0-18db55880958"
        >
          <p>
            <input
              className="order-input"
              min="1"
              onChange={[Function]}
              type="number"
              value={1}
            />
          </p>
          <p>
            Jollof Rice, Beef and Plantain
          </p>
          <p>
            1500.00
          </p>
          <Link
            className="remove-order"
            clickHandler={[Function]}
            id={null}
          >
            <button
              className="link-btn remove-order"
              id={null}
              onClick={[Function]}
            >
              ×
            </button>
          </Link>
        </div>
        <div
          className="order-amount"
        >
          <div>
            <p>
              Total:
            </p>
            <h2>
              ₦
              1500
            </h2>
          </div>
        </div>
      </div>
      <br />
    </div>
  </div>
</Cart>
`;

exports[`Cart renders correctly 1`] = `
<React.Fragment>
  <div
    className="side-cart-toggle d-none-xl"
  >
    <span
      aria-hidden="true"
      id="cart-toggler"
      onClick={[Function]}
    >
      <div
        className="text-center"
      >
        <img
          alt="cart"
          src="/img/cart.png"
        />
          
        <sup>
          1
        </sup>
      </div>
      <p
        className="text-center"
      >
        Total: ₦
        1500
      </p>
    </span>
    <span
      className="cart-btn"
    >
      <Link
        href="/order-confirmation"
        replace={false}
        to="/order-confirmation"
      >
        <button
          className="btn btn-pri btn-block checkout-btn"
        >
          Checkout
        </button>
      </Link>
    </span>
  </div>
  <div
    className="side-cart null"
    id="side-cart"
  >
    <div
      className="cart"
    >
      <div
        className="cart-header"
      >
        <h2>
          Your Cart
        </h2>
        <CloseIcon
          btnID="side-cart-close-icon"
          clickHandler={[Function]}
          divClass="d-none-xl"
        />
      </div>
      <hr />
      <div
        className="order-summary"
      >
        <div
          key="81211c24-51c0-46ec-b1e0-18db55880958"
        >
          <p>
            <input
              className="order-input"
              min="1"
              onChange={[Function]}
              type="number"
              value={1}
            />
          </p>
          <p>
            Jollof Rice, Beef and Plantain
          </p>
          <p>
            1500.00
          </p>
          <Link
            className="remove-order"
            clickHandler={[Function]}
            id={null}
          >
            ×
          </Link>
        </div>
        <div
          className="order-amount"
        >
          <div>
            <p>
              Total:
            </p>
            <h2>
              ₦
              1500
            </h2>
          </div>
        </div>
      </div>
      <br />
      <Link
        href="/order-confirmation"
        replace={false}
        to="/order-confirmation"
      >
        <button
          className="btn btn-pri btn-block checkout-btn"
        >
          Checkout
        </button>
      </Link>
    </div>
  </div>
</React.Fragment>
`;

exports[`Cart renders message when there is no order 1`] = `
<React.Fragment>
  <div
    className="side-cart-toggle d-none-xl"
  >
    <span
      aria-hidden="true"
      id="cart-toggler"
      onClick={[Function]}
    >
      <div
        className="text-center"
      >
        <img
          alt="cart"
          src="/img/cart.png"
        />
          
        <sup>
          0
        </sup>
      </div>
      <p
        className="text-center"
      >
        Total: ₦
        0
      </p>
    </span>
    <span
      className="cart-btn"
    >
      <Link
        href="/order-confirmation"
        replace={false}
        to="/order-confirmation"
      >
        <button
          className="btn btn-pri btn-block checkout-btn"
        >
          Checkout
        </button>
      </Link>
    </span>
  </div>
  <div
    className="side-cart null"
    id="side-cart"
  >
    <div
      className="cart"
    >
      <div
        className="cart-header"
      >
        <h2>
          Your Cart
        </h2>
        <CloseIcon
          btnID="side-cart-close-icon"
          clickHandler={[Function]}
          divClass="d-none-xl"
        />
      </div>
      <hr />
      <div
        className="empty-cart"
      >
        <p>
          Your Cart is Empty
        </p>
      </div>
      <br />
      <Link
        href="/order-confirmation"
        replace={false}
        to="/order-confirmation"
      >
        <button
          className="btn btn-pri btn-block checkout-btn"
        >
          Checkout
        </button>
      </Link>
    </div>
  </div>
</React.Fragment>
`;
